Catches error while sending notifications

jonasheinrich opened the pull request:
Fixes https://community.openproject.org/work_packages/19200

Commit:
Catches error while sending notifications

kgalli commented on config/locales/de.yml:
```diff
@@ -62,6 +62,7 @@ de:
   label_version: "Version"
 
   notice_successful_notification: "Benachrichtigung erfolgreich gesendet"
+  notice_notification_with_errors: "Benachrichtigung gesendet. Folgende Empfänger konnten nicht benachrichtigt werden: "
```
I suggest we use flash error to avoid confusion.

kgalli commented on config/locales/en.yml:
```diff
@@ -62,6 +62,7 @@ en:
   label_version: "Version"
 
   notice_successful_notification: "Notification sent successfully"
+  notice_notification_with_errors: "Notification sent. The following recipients could not be notified: "
```
Same as my comment above.

Commit:
Makes the notification to an error

kgalli committed:
Refactoring: reduce of unnecessary block chaining

kgalli committed:
Fix: use name instead of email

This fix is necessary because we do not want to
show sensitive data in this case the email of
users on screen.

kgalli committed:
Refactoring: increase readability

kgalli committed:
Refactoring: replacement of `and` with `&&`
